lib/sequencescape/order.rb

```diff
@@ -9,5 +9,6 @@ class Sequencescape::Order < ::Sequencescape::Api::Resource
   # It does, but we don't care for now.
   # has_many :request_types
 
-  attr_accessor :request_options
+  attribute_accessor :request_options
+  attribute_writer :autodetect_studies_projects
```

Comment:
OK so this means we can insert an order through the API, and set this attribute on it? Which will then trigger a load of stuff to happen?

Reply:
Yes. It means that it will try and work out the study and project based on the aliquots. In the case of project, it's not set, so we actually set that up as part of the submission template, but the study gets pulled directly from the aliquots. This means that if R&D want to stick some of their samples onto a plate going into the pipeline, it will just work.

We could get limber to extract the study information for us, and pass that through, but that get surprisingly expensive, especially as we don't know we need the information until we've already pulled it back.

The V2 API will be better for this, as it get to use ids, rather than uuids, which are a lot cheaper to retrieve. But still, there's an advantage to being able to tell the submission builder to just make the sensible choice.